src/org/openstreetmap/josm/gui/tagging/presets/TaggingPreset.java

```diff
@@ -634,4 +642,118 @@ public String getToolbarString() {
         ToolbarPreferences.ActionParser actionParser = new ToolbarPreferences.ActionParser(null);
         return actionParser.saveAction(new ToolbarPreferences.ActionDefinition(this));
     }
+
+    /**
+     * Get a string representing this preset as the search query that can
+     * be used in {@link SearchCompiler}.
+     * @return A query to search for {@link OsmPrimitive} that match this preset.
+     */
+    public String getSearchQuery() {
+        return queryGenerator.buildPresetSearchQuery(this);
+    }
+
+    /**
+     * This class encapsulates logic to convert any preset to a string that can be
+     * fed to {@link SearchCompiler} to search for OSM primitives.
+     */
+    private static class TaggingPresetSearchQueryGenerator {
+
+        /**
+         * keywords used to build the query.
+         */
+        private static final String TYPE = "type:";
+        private static final String WILD_CARD = "*";
+        private static final String OR = " | ";
+        private static final String AND = " ";
+        private static final String OPENING_PAR = "(";
+        private static final String CLOSING_PAR = ")";
+
+        /**
+         * types used to search for OSM objects.
+         */
+        private static final String WAY = "WAY";
+        private static final String NODE = "NODE";
+        private static final String RELATION = "RELATION";
```

Comment:
The search parser has the exact same strings for parsing - wouldn't it be best to use them?